### PR TITLE
Add support for ATtiny boards with the Micronucleus 'virtual usb' bootloader

### PR DESCRIPTION
**Pull request by ReinoutHeeck**
Adds a new Cargo runner 'micronucleus-runner' and working example for the Digispark board.

micronucleus-runner is a new crate -a brother to ravedude-. I don't know what the procedure is to introduce new crates into this project, the changes I made pretend it already exists on crates.io...

The code is trivial, the readme is bigger ;-)

### CODE REVIEW

**Comment by simonsso** [comment]
Hi @ReinoutHeeck is there anything the new micronucleus-runner does which the old ravedude does not do? 

The readme is a clue:
> Boards that have the Micronucleus 'virtual usb' boot loader require 
Is this something other than the default bootloader?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -116,3 +116,21 @@ jobs:
         with:
           command: check
           args: --manifest-path ravedude/Cargo.toml
+
+micronucleus_runner:
+    name: "micronucleus-runner"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
```
Comment (simonsso) [inline .github/workflows/ci.yml:125]:
```suggestion
        uses: actions/checkout@v3
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -116,3 +116,21 @@ jobs:
...
+      - name: Install Rust
+        uses: actions-rs/toolchain@v1
```
Comment (simonsso) [inline .github/workflows/ci.yml:127]: `action-rs/toolchain` is unmaintained.  `actions-rust-lang/setup-rust-toolchain@v1` is one alternative

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -116,3 +116,21 @@ jobs:
...
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+      - name: Check micronucleus-runner
+        uses: actions-rs/cargo@v1
```
Comment (simonsso) [inline .github/workflows/ci.yml:133]: `actions-rs/cargo` is unmaintained. Run cargo directly in the actions is a solution.

**Comment by ReinoutHeeck** [comment]
> is there anything the new micronucleus-runner does which the old ravedude does not do?

AvrDude/RaveDude require an external programmer to program the chip on the Digispark board, (there is no bootloader required in such a setup). 

The Micronucleus bootloader turns the Digispark into a self-programming board which makes for a very smooth entry into programming this device, particularly because it sold with this bootloader pre-installed.

As far as I know AvrDude does not support this programming protocol at all.

**Comment by ReinoutHeeck** [comment]
@simonsso 

Fredrik, thank you for your suggestions!

 I tried to refresh my branch on github and incorporate your suggestions but since my branch was not automatically mergable github rolled back all my changes and automatically marked my contribution as 'closed'. I spent some time trying to get a diff view on what I authored two months ago but failed miserably. This is not the first time I find git a friend and github a foe. 

Unfortunately I lack the time and energy to undo this mess, so my micronucleus contributions have today been killed by my tools :-(

Once again, thanks for your time

**Pull request closed without merging**